modules/core/log/src/AssetLogs.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace Drupal\farm_log;
+
+use Drupal\asset\Entity\AssetInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+
+/**
+ * Service for loading logs that reference assets.
+ *
+ * @internal
+ */
+class AssetLogs implements AssetLogsInterface {
+
+  /**
+   * Entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected EntityTypeManagerInterface $entityTypeManager;
+
+  /**
+   * Log query factory.
+   *
+   * @var \Drupal\farm_log\LogQueryFactoryInterface
+   */
+  protected LogQueryFactoryInterface $logQueryFactory;
+
+  /**
+   * Class constructor.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   Entity type manager.
+   * @param \Drupal\farm_log\LogQueryFactoryInterface $log_query_factory
+   *   Log query factory.
+   */
+  public function __construct(EntityTypeManagerInterface $entity_type_manager, LogQueryFactoryInterface $log_query_factory) {
+    $this->entityTypeManager = $entity_type_manager;
+    $this->logQueryFactory = $log_query_factory;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getLogs(AssetInterface $asset, string $log_type = NULL, bool $access_check = TRUE): array {
+    $log_ids = $this->query($asset, $log_type, $access_check)->execute();
+    return $this->entityTypeManager->getStorage('log')->loadMultiple($log_ids);
```

Comment:
I know `loadMutliple()` will load all entities if `NULL` is passed instead of an array of IDs. I'm not quite sure what the behavior is if an empty array is passed... I assume it won't load any? But maybe we could add a quick check and only `loadMultiple()` if we have some IDs?

Reply:
fixup commit: https://github.com/farmOS/farmOS/pull/850/commits/18172abd9099d65ffb5b09f1cecd0db0e439ac62